python/tests/core/metrics/test_condition_count.py

```diff
@@ -262,3 +263,65 @@ def test_condition_count_in_dataset_profile() -> None:
     assert summary["condition_count/total"] > 0
     assert summary["condition_count/alpha"] > 0
     assert summary["condition_count/digit"] > 0
+
+
+def _build_profile(data: List[int]) -> DatasetProfile:
+    """build up a "reference profile" to compare against"""
+    row = {"col1": data}
+    frame = pd.DataFrame(data=row)
+    prof = DatasetProfile()
+    prof.track(pandas=frame)  # track once to discover columns
+    prof.add_metric("col1", DistributionMetric.zero(MetricConfig()))
+    prof.track(pandas=frame)  # track again to populate the metric
+    return prof
+
+
+def test_profile_getter() -> None:
+    data = [1, 2, 3, 4, 5]
+    prof = _build_profile(data)
+    conditions = {
+        "above_min": Condition(X.greater_than(ProfileGetter(prof, "col1", "distribution/min"))),
+    }  # compare each logged value against profile's min
+    config = ConditionCountConfig(conditions=conditions)
+    metric = ConditionCountMetric.zero(config)
+    metric.columnar_update(PreprocessedColumn.apply(data))
+    summary = metric.to_summary_dict(None)
+    assert summary["total"] == len(data)
+    assert summary["above_min"] == len(data) - 1
+
+
+def test_metric_getter() -> None:
+    data = [1, 2, 3, 4, 5]
+    dist_metric = DistributionMetric.zero(MetricConfig())
+    dist_metric.columnar_update(PreprocessedColumn.apply(data))
+
+    conditions = {
+        "above_min": Condition(X.greater_than(MetricGetter(dist_metric, "min"))),
```

Comment:
Nicer syntax for instantiating `MetricGetter` and `ProfileGetter` ?

Reply:
It's okish. I think I would be more likely to type in Condition(X.greater_than, Metric Getter) then in condition do that. but that's just cause I'm used to sending in separate parameters.

Reply:
The `ValueGetter` (:> `MetricGetter`, `ProfileGetter`, `LiteralGetter`) has to be an argument to the `Predicate` rather than the `Condition` to support boolean compound predicates. E.g., `X.less_than(MetricGetter(distribution, "q_75").and_(X.greater_than(MetricGetter(distribution, "mean"))`.  The `Predicate`s can also be used in contexts where there are no `Condition`s.

Reply:
Ahh yes that make sense.